### ISSUE TITLE
Multiple value headers not matching

### ISSUE THREAD

**Issue by pmack24** [cjstehno/ersatz#65]
It seems that request with multiple value headers are not matched. Below is the expectation.

```groovy
server.expectations {
    get('/api/hello') {
        called 1
        header 'Accept', 'application/json'
        header 'Accept', 'application/vnd.company+json'
        responder {
            code 200
                content  'msg': 'World', 'application/vnd.company+json'
            }
        }
}
```

However, this does not seem to match. Looking at the code it appears that only the first value is compared:

```groovy
static RequestMatcher header(final String name, final Matcher<String> m) {
    new RequestMatcher(m, { ClientRequest cr -> cr.headers.getFirst(name) })
}
```

```groovy
public String getFirst(String headerName) {
    HeaderValues headerValues = getEntry(headerName);
    if (headerValues == null) return null;
    return headerValues.getFirst();
}
```

Can this be fixed, or would you prefer I send a PR ?

Thanks

**Comment by cjstehno** [comment]
Erg, GitHub didn't send me an email about this. I will look into this in the next few days.

**Comment by cjstehno** [comment]
I am currently working on this issue, but if you can't wait a few days a workaround is to define a matcher in the expectation:

```groovy
header 'Accept', Matchers.isOneOf('application/json', 'application/vnd.company+json')
```

A test would look something like this: 

```groovy
def 'multiple header matching support'() {
    setup:
    ersatzServer.expectations {
        get('/api/hello') {
            called 1
//                header 'Accept', 'application/json'
//                header 'Accept', 'application/vnd.company+json'
            header 'Accept', Matchers.isOneOf('application/json', 'application/vnd.company+json')
            responder {
                code 200
                content 'msg': 'World', 'application/vnd.company+json'
            }
        }
    }

    when:
    okhttp3.Response response = client.newCall(new okhttp3.Request.Builder().get().url(url('/api/hello'))
//            .header('Accept', 'application/json')
        .header('Accept', 'application/vnd.company+json')
        .build()
    ).execute()

    then:
    response.code() == 200
    response.body().string() == '[msg:World]'

    and:
    ersatzServer.verify()
}
```

At this point I also see this as how you would implement the case where your request will match at least one of the specified header. In your example above, the request would have to have ALL of the specified header values.

I will probably also provide other convenience methods for specifying more than one header value.

**Comment by cjstehno** [comment]
There is a fix for this in the `development` branch (1.5.0 release coming soon) - I will close the issue for now, but feel free to reopen it if you run into issues.
